### PR TITLE
AWS: Remove AssertHelpers usage

### PR DESCRIPTION
**Pull request by coded9**
Remove AssertHelpers usage as part of #7094

### CODE REVIEW

File: aws/src/integration/java/org/apache/iceberg/aws/dynamodb/TestDynamoDbCatalog.java [hunk 1]
```diff
@@ -243,15 +234,17 @@ public void testDropTable() {
...
+            () ->
+                s3.headObject(
+                    HeadObjectRequest.builder()
+                        .bucket(testBucket)
+                        .key(
+                            metadataLocation.substring(
+                                testBucket.length() + 6)) // s3:// + end slash
+                        .build()))
+        .as("metadata location should be deleted")
+        .isInstanceOf(NoSuchKeyException.class);
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/dynamodb/TestDynamoDbCatalog.java:247]: we need a `hasMessageXyZ(...)` check here.

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 2]
```diff
@@ -426,10 +408,8 @@ public void testS3Exception() {
 
     GlueTableOperations spyOps = Mockito.spy(ops);
     failCommitAndThrowException(spyOps, S3Exception.builder().statusCode(300).build());
-
-    AssertHelpers.assertThrows(
-        null, S3Exception.class, () -> spyOps.commit(metadataV2, metadataV1));
-
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .isInstanceOf(S3Exception.class);
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:412]: same here, we should add a check against the message

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 3]
```diff
@@ -446,9 +426,8 @@ public void testOtherGlueException() {
 
     GlueTableOperations spyOps = Mockito.spy(ops);
     failCommitAndThrowException(spyOps, GlueException.builder().statusCode(300).build());
-
-    AssertHelpers.assertThrows(
-        null, GlueException.class, () -> spyOps.commit(metadataV2, metadataV1));
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .isInstanceOf(GlueException.class);
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:430]: and here

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 4]
```diff
@@ -466,9 +445,8 @@ public void testInternalServerErrorRetryCommit() {
 
     GlueTableOperations spyOps = Mockito.spy(ops);
     failCommitAndThrowException(spyOps, GlueException.builder().statusCode(500).build());
-
-    AssertHelpers.assertThrows(
-        null, CommitFailedException.class, () -> spyOps.commit(metadataV2, metadataV1));
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .isInstanceOf(CommitFailedException.class);
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:449]: here as well

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java (hunk 2 above)
Comment (coded9) [reply aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:412]: I think here it's not needed as the exception without a message is thrown from the above line

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java (hunk 3 above)
Comment (coded9) [reply aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:430]: I think here it's not needed as the exception without a message is thrown from the above line

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java (hunk 2 above)
Comment (nastra) [reply aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:412]: if there's no message, then we should still add a check for whatever the message is (null, empty)

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 5]
```diff
@@ -357,12 +345,10 @@ public void testCreateTableWithInvalidDB() {
...
     GlueTableOperations spyOps = Mockito.spy(ops);
     failCommitAndThrowException(spyOps, EntityNotFoundException.builder().build());
-
-    AssertHelpers.assertThrows(
-        "Should throw not found exception",
-        NotFoundException.class,
-        "because Glue cannot find the requested entity",
-        () -> spyOps.commit(metadataV2, metadataV1));
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .as("Should throw not found exception")
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:349]: not needed

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 6]
```diff
@@ -380,13 +366,10 @@ public void testGlueAccessDeniedException() {
...
     failCommitAndThrowException(spyOps, AccessDeniedException.builder().build());
-
-    AssertHelpers.assertThrows(
-        "Should throw forbidden exception",
-        ForbiddenException.class,
-        "because Glue cannot access the requested resources",
-        () -> spyOps.commit(metadataV2, metadataV1));
-
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .as("Should throw forbidden exception")
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:370]: not needed

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 7]
```diff
@@ -403,12 +386,11 @@ public void testGlueValidationException() {
...
     GlueTableOperations spyOps = Mockito.spy(ops);
     failCommitAndThrowException(spyOps, ValidationException.builder().build());
-
-    AssertHelpers.assertThrows(
-        "Should throw validation exception",
-        org.apache.iceberg.exceptions.ValidationException.class,
-        "because Glue encountered a validation exception while accessing requested resources",
-        () -> spyOps.commit(metadataV2, metadataV1));
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .as("Should throw validation exception")
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:390]: not needed

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 8]
```diff
@@ -265,13 +256,10 @@ public void testSucceededCommitThrowsUnknownException() {
...
     breakFallbackCatalogCommitCheck(spyOps);
-
-    AssertHelpers.assertThrows(
-        "Should throw CommitStateUnknownException since the catalog check was blocked",
-        CommitStateUnknownException.class,
-        "Datacenter on fire",
-        () -> spyOps.commit(metadataV2, metadataV1));
-
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .as("Should throw CommitStateUnknownException since the catalog check was blocked")
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:260]: not needed

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 9]
```diff
@@ -236,12 +229,10 @@ public void testFailedCommitThrowsUnknownExceptionWhenStatusCheckFails() {
...
     failCommitAndThrowException(spyOps);
     breakFallbackCatalogCommitCheck(spyOps);
-
-    AssertHelpers.assertThrows(
-        "Should throw CommitStateUnknownException since the catalog check was blocked",
-        CommitStateUnknownException.class,
-        "Datacenter on fire",
-        () -> spyOps.commit(metadataV2, metadataV1));
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .as("Should throw CommitStateUnknownException since the catalog check was blocked")
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:233]: not needed

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java [hunk 10]
```diff
@@ -84,12 +81,10 @@ public void testFailedCommitThrowsUnknownException() {
...
     GlueTableOperations spyOps = Mockito.spy(ops);
     failCommitAndThrowException(spyOps);
-
-    AssertHelpers.assertThrows(
-        "Should throw CommitStateUnknownException since exception is unexpected",
-        CommitStateUnknownException.class,
-        "Datacenter on fire",
-        () -> spyOps.commit(metadataV2, metadataV1));
+    Assertions.assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .as("Should throw CommitStateUnknownException since exception is unexpected")
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java:85]: not needed

File: aws/src/integration/java/org/apache/iceberg/aws/dynamodb/TestDynamoDbCatalog.java (hunk 1 above)
Comment (coded9) [reply aws/src/integration/java/org/apache/iceberg/aws/dynamodb/TestDynamoDbCatalog.java]: added

File: aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java (hunk 2 above)
Comment (coded9) [reply aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogCommitFailure.java]: Thanks for the suggestion. Added

**Review by nastra — APPROVED** [review]

**Pull request merged**